holidays/countries/malaysia.py

```diff
@@ -136,259 +150,428 @@ def __init__(self, *args, **kwargs):
         StaticHolidays.__init__(self, cls=MalaysiaStaticHolidays)
         kwargs.setdefault("observed_rule", SUN_TO_NEXT_WORKDAY)
         super().__init__(*args, **kwargs)
+        self.dts_observed = set()
 
     def _populate_public_holidays(self):
-        dts_observed = set()
+        if self._year <= 1951:
+            return None
 
-        # Chinese New Year (one day in the States of Kelantan and Terengganu,
-        # two days in the other States).
-        dts_observed.add(self._add_chinese_new_years_day("Chinese New Year"))
-        # The second day of Chinese New Year is not a federal holiday in
-        # Kelantan and Terengganu. However, it is gazetted as a state holiday
-        # in both states, effectively making it a nationwide holiday.
-        dts_observed.add(self._add_chinese_new_years_day_two("Chinese New Year Holiday"))
+        self.dts_observed = set()
```

Comment:
There is an extra empty line which needs to be removed too
```suggestion
```

Reply:
`self.dts_observed` should be reset at each `_populate_public_holidays()` pass, for correct placement of "Eid al-Adha (observed)" in 2007 (early January, observed from 2006-12-31). Now it's made as `special_public_holidays_observed` in relevant subdivisions. If we remove `self.dts_observed = set()`, this holiday will be created by `self._populate_observed()` when `self.years` contains **both 2006 and 2007**, and will not be created when `self.years` **doesn't contain 2006**.

Reply:
Gotcha! Let's add a comment then, it'll be 100% useful when reading this code next time.